eth/taiko_api_backend.go

```diff
@@ -89,3 +92,44 @@ func (s *TaikoAPIBackend) TxPoolContent(
 		maxTransactionsLists,
 	)
 }
+
+// Get L2ParentHashes retrieves the preceding 256 parent hashes given a block number.
+func (s *TaikoAPIBackend) GetL2ParentHashes(blockID uint64) ([]common.Hash, error) {
+	var hashes []common.Hash
+	furthest := 256
+
+	if blockID < 257 {
+		furthest = int(blockID)
+	}
+
+	for i := 1; i < furthest; i++ {
+		block, err := s.eth.APIBackend.BlockByNumber(context.Background(), rpc.BlockNumber(blockID-uint64(i)))
+		if err != nil {
+			return nil, err
+		}
+
+		hashes = append(hashes, block.Hash())
+	}
+	return hashes, nil
+}
+
+// Get L2ParentBlocks retrieves the preceding 256 parent blocks given a block number.
+func (s *TaikoAPIBackend) GetL2ParentBlocks(blockID uint64) ([]map[string]interface{}, error) {
+	var parents []map[string]interface{}
+	b := ethapi.NewBlockChainAPI(s.eth.APIBackend)
+	furthest := 256
+
+	if blockID < 257 {
+		furthest = int(blockID)
+	}
+
+	for i := 1; i < furthest; i++ {
+		block, err := b.GetBlockByNumber(context.Background(), rpc.BlockNumber(blockID-uint64(i)), false)
+		if err != nil {
+			return nil, err
+		}
+		parents = append(parents, block)
+	}
+
+	return parents, nil
```

Comment:
Maybe I will just write something like this (seems we only need 4 lines of code?):
```go
        var headers []*types.Header
	for i := blockID; i == 0 && (blockID-i >= 256); i-- {
		headers = append(headers, s.eth.blockchain.GetHeaderByNumber(i))
	}
	return headers, nil
```